protons/driver.py

```diff
@@ -754,6 +755,110 @@ def _cache_force(self, titration_group_index, titration_state_index):
 
         self.titrationGroups[titration_group_index]['titration_states'][titration_state_index]['forces'] = f_params
 
+    def _ncmc_ghmc(self, context, titration_group_indices, initial_titration_states, final_titration_states):
+        """
+        Non equilibrium candidate Monte Carlo (NCMC) with a generalized Hamiltonian Monte Carlo (GHMC) propagator.
+
+        Notes
+        -----
+        The GHMC integrator is an simtk.openmm.CustomIntegrator object that calculates the protocol work internally.
+        Although it appears to only take one propagation step per perturbation step (i.e. ghmc.step(1)), multiple
+        propagation steps can be made depending on how the GHMC integrator was initialized.
+
+        To ensure the NCMC protocol is time symmetric, it has the form
+            propagation --> perturbation --> propagation
+
+        Parameters
+        ----------
+        context : simtk.openmm.Context
+            The context in which NCMC will be performed.
+
+        titration_group_indices :
+            The indices of the titratable groups that will be perturbed
+
+        initial_titration_states :
+            The initial protonation state of the titration groups
+
+        final_titration_states :
+            The final protonation state of the titration groups
+
+        Returns
+        -------
+        work : float
+          the protocol work of the NCMC procedure in multiples of kT.
+        """
+        # Turn the barostat and center of mass remover off, otherwise they contribute to the work
+        if self.barostat is not None:
+            self.barostat.setFrequency(0)
+        if self.cm_remover is not None:
+            self.cm_remover.setFrequency(0)
+
+        #TODO: remove explicit calculation of work with `getEnergy`
```

Comment:
Is this already addressed? I don't see `getEnergy` being called in this part of the code.